src/pages/list/search/articles/components/TagSelect/index.tsx

```diff
@@ -0,0 +1,173 @@
+import { Icon as LegacyIcon } from '@ant-design/compatible';
+import { Tag } from 'antd';
+import React, { Component } from 'react';
+
+import classNames from 'classnames';
+import styles from './index.less';
+
+const { CheckableTag } = Tag;
+
+export interface TagSelectOptionProps {
+  value?: string | number;
+  style?: React.CSSProperties;
+  checked?: boolean;
+  onChange?: (value: string | number | undefined, state: boolean) => void;
+}
+
+export interface TagSelectProps {
+  onChange?: (value: (string | number)[]) => void;
+  expandable?: boolean;
+  value?: (string | number)[];
+  defaultValue?: (string | number)[];
+  style?: React.CSSProperties;
+  hideCheckAll?: boolean;
+  actionsText?: {
+    expandText?: React.ReactNode;
+    collapseText?: React.ReactNode;
+    selectAllText?: React.ReactNode;
+  };
+  className?: string;
+  Option?: TagSelectOptionProps;
+  children?: React.ReactElement<TagSelectOption> | React.ReactElement<TagSelectOption>[];
+}
+
+const TagSelectOption: React.FC<TagSelectOptionProps> & {
+  isTagSelectOption: boolean;
+} = ({ children, checked, onChange, value }) => (
+  <CheckableTag
+    checked={!!checked}
+    key={value}
+    onChange={state => onChange && onChange(value, state)}
+  >
+    {children}
+  </CheckableTag>
+);
+
+TagSelectOption.isTagSelectOption = true;
+
+interface TagSelectState {
+  expand: boolean;
+  value: (string | number)[];
+}
+
+class TagSelect extends Component<TagSelectProps, TagSelectState> {
+  static defaultProps = {
+    hideCheckAll: false,
+    actionsText: {
+      expandText: '展开',
+      collapseText: '收起',
+      selectAllText: '全部',
+    },
+  };
+
+  static Option: TagSelectOption = TagSelectOption;
+
+  static getDerivedStateFromProps(nextProps: TagSelectProps) {
+    if ('value' in nextProps) {
+      return { value: nextProps.value || [] };
+    }
+    return null;
+  }
+
+  constructor(props: TagSelectProps) {
+    super(props);
+    this.state = {
+      expand: false,
+      value: props.value || props.defaultValue || [],
+    };
+  }
+
+  onChange = (value: (string | number)[]) => {
+    const { onChange } = this.props;
+    if (!('value' in this.props)) {
+      this.setState({ value });
+    }
+    if (onChange) {
+      onChange(value);
+    }
+  };
+
+  onSelectAll = (checked: boolean) => {
+    let checkedTags: (string | number)[] = [];
+    if (checked) {
+      checkedTags = this.getAllTags();
+    }
+    this.onChange(checkedTags);
+  };
+
+  getAllTags() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class method 'getAllTags' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)